litellm/llms/anthropic/cost_calculation.py

```diff
@@ -50,7 +50,7 @@ def cost_per_token(model: str, usage: Usage) -> Tuple[float, float]:
     _cache_creation_input_token_cost = model_info.get("cache_creation_input_token_cost")
     if _cache_creation_input_token_cost is not None:
         prompt_cost += (
-            float(usage._cache_creation_input_tokens) * _cache_creation_input_token_cost
+            float(usage._cache_creation_input_tokens) * (_cache_creation_input_token_cost - model_info["input_cost_per_token"])
         )
```

Comment:
this does not make sense to me, why are you making this change @Gullesnuffs. Can you link to docs on Anthropic that indicate you should do this

Reply:
https://docs.anthropic.com/en/docs/build-with-claude/prompt-caching

Specifically, the docs say "Cache write tokens are 25% more expensive than base input tokens". Litellm currently reports these tokens as 125% more expensive due to double counting, and this PR fixes the bug.

Reply:
@Gullesnuffs you are right. I also got noticed that completion_cost from litellm is too expensive for prompt-caching-enabled cases.
As you can see, SUM(spend) is $8.879 which is larger than $6.26 in my daily cost.
When I subtract prompt_cost for the requests which writes cache (which are over $0.5), it is super near to it. The modified one is the right column in the excel. The sum of modified version is $6.169
Of course you cannot believe this result because the target date is just today, but the logic worked for other days.
LiteLLM's implementation adds 125% of cost, instead of 25% of cost.
<img width="898" alt="litellm_cost" src="https://github.com/user-attachments/assets/a9e3f8e4-8bab-41e7-b2aa-2c099b03b184">

Reply:
Therefore, @Gullesnuffs fixed it with the least change, simply cancel the effect of prompt_cost for those cache_write_tokens.
It will be better to reorganize the conditional flow for cache_write cost.